R/util.R

```diff
@@ -181,3 +181,24 @@ key_value_collector <- function(init = list()) {
        get_all = function() env$res,
        get = function(keys) unlist(lapply(keys, get), recursive = FALSE))
 }
+
+ordered_map_to_list <- function(x) {
```

Comment:
You might move all the new functions introduced here into a single file so that it's easier to back out in one hit